src/benchmarks/micro/runtime/BenchmarksGame/fannkuch-redux-9.cs

```diff
@@ -0,0 +1,185 @@
+                var next = p[1];
+                p[0] = next;
+                for (int j = 1; j < i;) p[j] = p[++j];
+                p[i] = first;
+                first = next;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static void Copy(short* p, short* pp, int n)
+        {
+            var startL = (long*)p;
+            var stateL = (long*)pp;
+            var lengthL = n / 4;
+            int i = 0;
+            for (; i < lengthL; i++)
+            {
+                stateL[i] = startL[i];
+            }
+            for (i = lengthL * 4; i < n; i++)
+            {
+                pp[i] = p[i];
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static int CountFlips(short* p, short* pp, int n)
+        {
+            int flips = 1;
+            int first = *p;
+            short temp;
+            if (p[first] != 0)
+            {
+                Copy(p, pp, n);
+                do
+                {
+                    ++flips;
+                    if (first > 2)
+                    {
+                        short* lo = pp + 1, hi = pp + first - 1;
+                        do
+                        {
+                            temp = *lo;
+                            *lo = *hi;
+                            *hi = temp;
+                        } while (++lo < --hi);
+                    }
+                    temp = pp[first];
+                    pp[first] = (short)first;
+                    first = temp;
+                } while (pp[first] != 0);
+            }
+            return flips;
+        }
+
+        static void Run(int n, int taskSize)
+        {
+            int* count = stackalloc int[n];
+            int taskId, chksum = 0, maxflips = 0;
+            short* p = stackalloc short[n];
+            short* pp = stackalloc short[n];
+            while ((taskId = Interlocked.Decrement(ref taskCount)) >= 0)
+            {
+                FirstPermutation(p, pp, count, n, taskId * taskSize);
+                if (*p != 0)
+                {
+                    var flips = CountFlips(p, pp, n);
+                    chksum += flips;
+                    if (flips > maxflips) maxflips = flips;
+                }
+                for (int i = 1; i < taskSize; i++)
+                {
+                    NextPermutation(p, count);
+                    if (*p != 0)
+                    {
+                        var flips = CountFlips(p, pp, n);
+                        chksum += (1 - (i & 1) * 2) * flips;
+                        if (flips > maxflips) maxflips = flips;
+                    }
+                }
+            }
+            chkSums[-taskId - 1] = chksum;
+            maxFlips[-taskId - 1] = maxflips;
+        }
+
+        // Official runs use [Arguments(12, 3968050)] which takes ~4.2 sec vs ~330ms for 11
+        [Benchmark(Description = nameof(FannkuchRedux_9))]
+        [Arguments(11, 556355)]
```

Comment:
we could use `10` here as other `FannkuchRedux` are using it

https://github.com/dotnet/performance/blob/d6ff9585a295f0b2d7570b54d84c24607480d661/src/benchmarks/micro/runtime/BenchmarksGame/fannkuch-redux-2.cs#L24-L25

https://github.com/dotnet/performance/blob/d6ff9585a295f0b2d7570b54d84c24607480d661/src/benchmarks/micro/runtime/BenchmarksGame/fannkuch-redux-5.cs#L106-L107

```suggestion
        [Arguments(10, 556355)]
```